src/main/scala/chisel3/experimental/inlinetest/InlineTest.scala

```diff
@@ -88,23 +90,46 @@ object TestHarnessGenerator {
   }
 }
 
+private class TestGenerator[M <: RawModule, R](
+  /** The index of this test in the order they were declared. */
+  val index: Int,
```

Comment:
The PR can accomplish the same thing (preservation of test insertion order) by using a `LinkedHashMap` instead of `HashMap`.